### PR TITLE
ENH: Automatic generation of template citations

### PR DESCRIPTION
**Pull request by mgxd**
This is a proof of concept of #33 

This requires a few changes to existing template descriptions:
- Change `ReferencesAndLinks` type from list to dictionary to facilitate extraction
```
"ReferencesAndLinks": {
    "1": "https://doi.org/10.1093/cercor/bhr291",
    "2": "https://github.com/Washington-University/HCPpipelines/tree/master/global/templates"
  }
```
- Standardizing `HowToAcknowledge` fields to include curly brackets with Reference IDs
```
"HowToAcknowledge": "Reference the following entries in ReferencesAndLinks: {1, 2}"
```

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -23,6 +23,7 @@ install_requires =
     pybids >= 0.9.2
     requests
     tqdm
+    doi2bib
```
Comment (mgxd) [inline setup.cfg]: Do we want to have this in the base installation or as an extra?

**Comment by oesteban** [comment]
Sounds like a reasonable extra to me

**Comment by mgxd** [comment]
Okay - this is ready for review

File: templateflow/api.py [hunk 2]
```diff
@@ -146,6 +148,47 @@ def get_metadata(template):
     return loads(filepath.read_text())
 
 
+def get_citations(template, bibtex=False):
```
Comment (oesteban) [inline templateflow/api.py]: When `bibtex=True`, it should return the URL if it is not a doi.

File: templateflow/tests/test_api.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+and C. Robert Almli and Robert C. McKinstry and D. Louis Collins},\
+\n\ttitle = {Unbiased average age-appropriate atlases for pediatric studies},\
+\n\tjournal = {{NeuroImage}}\n}\n\
+@article{Fonov_2009,\n\tdoi = {10.1016/s1053-8119(09)70884-5},\
+\n\turl = {https://doi.org/10.1016%2Fs1053-8119%2809%2970884-5},\
+\n\tyear = 2009,\n\tmonth = {jul},\n\tpublisher = {Elsevier {BV}},\
+\n\tvolume = {47},\n\tpages = {S102},\
+\n\tauthor = {VS Fonov and AC Evans and RC McKinstry and CR Almli and DL Collins},\
+\n\ttitle = {Unbiased nonlinear average age-appropriate brain templates from birth to adulthood},\
+\n\tjournal = {{NeuroImage}}\n}"""
```
Comment (oesteban) [inline templateflow/tests/test_api.py]:
```suggestion
    assert bibs[0] == """\
@article{Fonov_2011,\n\tdoi = {10.1016/j.neuroimage.2010.07.033},\
\n\turl = {https://doi.org/10.1016%2Fj.neuroimage.2010.07.033},\
\n\tyear = 2011,\n\tmonth = {jan},\n\tpublisher = {Elsevier {BV}},\
\n\tvolume = {54},\n\tnumber = {1},\n\tpages = {313--327},\
\n\tauthor = {Vladimir Fonov and Alan C. Evans and Kelly Botteron \
and C. Robert Almli and Robert C. McKinstry and D. Louis Collins},\
\n\ttitle = {Unbiased average age-appropriate atlases for pediatric studies},\
\n\tjournal = {{NeuroImage}}\n}"""
    assert bibs[1] == """\
@article{Fonov_2009,\n\tdoi = {10.1016/s1053-8119(09)70884-5},\
\n\turl = {https://doi.org/10.1016%2Fs1053-8119%2809%2970884-5},\
\n\tyear = 2009,\n\tmonth = {jul},\n\tpublisher = {Elsevier {BV}},\
\n\tvolume = {47},\n\tpages = {S102},\
\n\tauthor = {VS Fonov and AC Evans and RC McKinstry and CR Almli and DL Collins},\
\n\ttitle = {Unbiased nonlinear average age-appropriate brain templates from birth to adulthood},\
\n\tjournal = {{NeuroImage}}\n}"""
```

File: templateflow/tests/test_api.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+def test_citations(tmp_path):
+    cits = api.get_citations('MNI152NLin2009cAsym')
+    assert cits == [
+        'https://doi.org/10.1016/j.neuroimage.2010.07.033',
+        'https://doi.org/10.1016/S1053-8119(09)70884-5'
+    ]
+
+    bibs = api.get_citations('MNI152NLin2009cAsym', bibtex=True)
+    assert bibs == """\
+@article{Fonov_2011,\n\tdoi = {10.1016/j.neuroimage.2010.07.033},\
```
Comment (oesteban) [inline templateflow/tests/test_api.py]: I would also force the id of each reference to be the identifier of the template, as in:
```suggestion
@article{mni152nlin2009casym0,\n\tdoi = {10.1016/j.neuroimage.2010.07.033},\
```

`mni152nlin2009casym0`, `mni152nlin2009casym1` and so on, so forth.

File: templateflow/tests/test_api.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
+from .. import api
+
+
```
Comment (oesteban) [inline templateflow/tests/test_api.py]: Please add tests (maybe parametrized?):
- With only URLs (no dois)
- Mixing urls and dois

File: setup.cfg [hunk 6]
```diff
@@ -41,9 +41,10 @@ exclude =
     *.tests
 
 [options.extras_require]
+citations =
```
[the remark this answers is not in the document]
Comment (mgxd) [reply setup.cfg]: yes

**Comment by mgxd** [comment]
Once we drop 3.5 fully (#36), things should be working.

File: templateflow/api.py [hunk 7]
```diff
@@ -146,6 +149,29 @@ def get_metadata(template):
...
+    Parameters
+    ----------
+    template : :obj:`str`
+        A template identifier (e.g., ``MNI152NLin2009cAsym``).
+    bibtex : :obj:`bool`, optional
+        Generate citations in BibTeX format.
+
+    """
+    data = get_metadata(template)
+    refs = data.get('ReferencesAndLinks', {})
```
Comment (oesteban) [inline templateflow/api.py]: Can we keep `ReferencesAndLinks` as a list, to be consistent with BIDS?

**Comment by mgxd** [comment]
@oesteban did the skeleton change between the last 2 commits?

**Comment by oesteban** [comment]
I was testing, so that's possible.

**Comment by oesteban** [comment]
I would get #39 merged first to avoid this kind of problems

**Pull request merged**